python/ray/util/client/server/proxier.py

```diff
@@ -273,12 +273,12 @@ def start_specific_server(self, client_id: str,
         output, error = self.node.get_log_file_handles(
             f"ray_client_server_{specific_server.port}", unique=True)
 
-        serialized_runtime_env = job_config.get_serialized_runtime_env()
-        if serialized_runtime_env == "{}":
+        runtime_env = job_config.runtime_env
+        if not runtime_env:
```

Comment:
Should we also go into this case if `runtime_env == {}`?  In `JobConfig.__init__` and `JobConfig.set_runtime_env()` we have `self.runtime_env = runtime_env or dict()`, so this would prevent one RPC to the agent in the common case where there's no runtime env.  Maybe this makes the code less clean though, or maybe the right thing is to get rid of the `or dict()`.

Reply:
This should already handle the empty dict case:
```python
>>> not {}
True
```